### PR TITLE
fix: Add port to locally generated url for Event

### LINKED ISSUE #1162: Locally generated URLs for events does not take port into account
**Describe the bug**
When using local deployment, url for events consists of only localhost and port is missing which leads us to wrong url.

**To Reproduce**
1. Create an event (if you don't have one)
2. Click on view event. 

You will see this in the general info about the event

**Expected behavior**
It should contain port number when using local deployment so that we do not have to manually enter the port to see events page.

**Additional context**
URL is a computed property of the event model calculated with this function
``` JS
url: computed('identifier', function() {
    return `${location.protocol}//${location.hostname}${this.get('router').urlFor('public', this.get('id'))}`;
  }),
```
**Originally opened here: https://github.com/fossasia/open-event-server/issues/4733**

### PR DESCRIPTION
**Pull request by srv-twry**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds port to locally generated URL for Event.

#### Changes proposed in this pull request:
URL is a computed property of the Event model. For local setup, adds the port number to the url.

Fixes #1162

### CODE REVIEW

**Comment by srv-twry** [comment]
fixed_url

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,3 @@
 GOOGLE_API_KEY="Sample Key"
 API_HOST=http://localhost:5000
+LOCAL=true
```
Comment (srv-twry) [inline .env.example]: The issue with this solution is that if the developer doesn't specify LOCAL as true in the env file or the env file doesn't exist, then the port won't be added to the generated url. 
I couldn't figure out any other approach. Open for alternative approach.

Comment (niranjan94) [reply .env.example]: This is not needed at all. You can just look for `location.port`. If you get a value, use it. Else don't

File: app/models/event.js [hunk 2]
```diff
@@ -153,6 +153,9 @@ export default ModelBase.extend(CustomPrimaryKeyMixin, {
   }),
 
   url: computed('identifier', function() {
+    if (location.port) {
+      return `${location.protocol}//${location.hostname}:${location.port}${this.get('router').urlFor('public', this.get('id'))}`;
+    }
```
Comment (niranjan94) [inline app/models/event.js]: Actually, ... forgot about `location.origin`.

File: app/models/event.js [hunk 3]
```diff
@@ -153,6 +153,9 @@ export default ModelBase.extend(CustomPrimaryKeyMixin, {
   }),
 
   url: computed('identifier', function() {
+    if (location.port) {
+      return `${location.protocol}//${location.hostname}:${location.port}${this.get('router').urlFor('public', this.get('id'))}`;
```
Comment (niranjan94) [inline app/models/event.js]: Actually, forgot about `location.origin`.

Comment (niranjan94) [inline app/models/event.js]: Actually, forgot about `location.origin`. 

```js
return location.origin + this.get('router').urlFor('public', this.get('id'));
```

Comment (niranjan94) [reply app/models/event.js]: Just use this. So, we don't need to worry about port.

**Pull request merged**